main.tf

```diff
@@ -88,9 +88,49 @@ resource "aws_instance" "this" {
       "Name" = var.instance_count > 1 || var.use_num_suffix ? format("%s${var.num_suffix_format}", var.name, count.index + 1) : var.name
     },
     var.volume_tags,
+    var.tags,
   )
 
   credit_specification {
     cpu_credits = local.is_t_instance_type ? var.cpu_credits : null
   }
+
+  lifecycle {
```

Comment:
Is this really necessary ?

Reply:
Yes
To avoid clobbering the tags aws-instance<->ebs volume
https://github.com/hashicorp/terraform-provider-aws/issues/770

Reply:
I should point out I have made this change before and had it rejected on the basis it breaks backwards compatibility, it will certainly avoid clobbering the tags, but in this instance I was told to make a private fork.